### PR TITLE
fix: call `UnregisterIsolate` consistently

### PR DESCRIPTION
**Pull request by indutny**
#### Description of Change

`JavascriptEnvironment` is the class that calls `RegisterIsolate()`
so it should be the one to call `UnregisterIsolate`, and this can happen
right before disposing the aforementioned `isolate`.

See: https://github.com/electron/electron/pull/28468

#### Checklist

cc @deepak1556 

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [ ] relevant documentation is changed or added
- [ ] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: no-notes

### CODE REVIEW

File: shell/browser/javascript_environment.cc [hunk 1]
```diff
@@ -167,6 +167,11 @@ JavascriptEnvironment::~JavascriptEnvironment() {
   }
   isolate_->Exit();
   g_isolate = nullptr;
+
+  DCHECK_NE(platform_, nullptr);
+  platform_->DrainTasks(isolate_);
```
Comment (deepak1556) [inline shell/browser/javascript_environment.cc:172]: Since we are running tasks, should this be moved before the context and isolate are exited.

Comment (indutny) [reply shell/browser/javascript_environment.cc]: Force pushed. PTAL

**Review by deepak1556 — APPROVED** [review]

**Review by codebytere — APPROVED** [review]

**Review by nornagon — CHANGES_REQUESTED** [review]
Should this have some release notes? Is there any observable effect from this change? Does it fix any crash?

**Comment by indutny** [comment]
@nornagon there is no observable effect AFAIK. I think it makes sense to add release notes to the [other PR](https://github.com/electron/electron/pull/28471) that got landed in 11-x-y since it actually fixes the worker_threads crash in v11.

File: shell/browser/javascript_environment.cc [hunk 2]
```diff
@@ -160,13 +160,19 @@ JavascriptEnvironment::JavascriptEnvironment(uv_loop_t* event_loop)
...
+
   {
     v8::Locker locker(isolate_);
     v8::HandleScope scope(isolate_);
     context_.Get(isolate_)->Exit();
   }
   isolate_->Exit();
   g_isolate = nullptr;
+
+  DCHECK_NE(platform_, nullptr);
```
Comment (nornagon) [inline shell/browser/javascript_environment.cc:173]: is this `DCHECK` checking that `Context::Exit` or `Isolate::Exit` didn't null out `platform_`? If not, then should it perhaps be at the beginning of the function, before the first reference to `platform_`?

Comment (indutny) [reply shell/browser/javascript_environment.cc]: You're absolutely right. This check has to be at the top. I was addressing another review feedback and forgot to move it up with that statement. It is resolved now, though!

**Review by nornagon — APPROVED** [review]

**Comment by indutny-signal** [comment]
Should this be merged cc @nornagon @deepak1556 @codebytere ?

**Comment by indutny** [comment]
Awesome. Thanks a lot!

**Pull request merged**